polkadot/runtime/parachains/src/runtime_api_impl/vstaging.rs

```diff
@@ -16,18 +16,28 @@
 
 //! Put implementations of functions from staging APIs here.
 
-use primitives::{
-	ValidatorIndex,
-};
-use sp_std::prelude::*;
+use crate::shared;
+use primitives::ValidatorIndex;
+use sp_std::prelude::Vec;
+use sp_std::collections::btree_map::BTreeMap;
 
 /// Implementation for `DisabledValidators`
-pub fn disabled_validators<T: pallet_session::Config>() -> Vec<ValidatorIndex> {
-	// <pallet_session::Pallet<T>>::disabled_validators()
-	// 	.iter()
-	// 	.cloned()
-	// 	.map(|v| ValidatorIndex(v))
-	// 	.collect()
-	// TODO
-	Vec::new()
+pub fn disabled_validators<T>() -> Vec<ValidatorIndex>
+where
+	T: pallet_session::Config + shared::Config,
+{
+	let shuffled_indices = <shared::Pallet<T>>::active_validator_indices();
+	// mapping from raw validator index to `ValidatorIndex`
```

Comment:
`ValidatorIndex` in polkadot is an index in active validators array while [`ValidatorIndex` in substrate](https://github.com/paritytech/polkadot-sdk/blob/5174b9d2d7ae67ec05a481508e7ec275e4d4f1d9/polkadot/runtime/parachains/src/shared.rs#L143) is completely different index in `all_validators` ?

Or put in other way:
1. [0....N] - all validator indecies (in substrate)
2. The array above gets shuffled and truncated to become `active_validator_indices` which is still 'substrate indexes' [here](https://github.com/paritytech/polkadot-sdk/blob/5174b9d2d7ae67ec05a481508e7ec275e4d4f1d9/polkadot/runtime/parachains/src/shared.rs#L194))
3. [`ValidatorIndex` in polkadot](https://github.com/paritytech/polkadot-sdk/blob/7cbe0c76ef8fd2aabf9f07de0156941ce3ed44b0/polkadot/primitives/src/v6/mod.rs#L115) is an index within `active_validator_indecies`. It is one abstraction above the substrate indecies.

Is this correct?

Reply:
Yes. What is confusing is that `active_validator_indices` holds a vec of `ValidatorIndex`, whereas it should be some other type representing raw substrate index. We use `ValidatorIndex` to represent an index in that vec, so it's in range of 0..max_validators.

Reply:
Oh I see... Thanks for the explanation.